### ISSUE TITLE
Citations are always shown as "referenced"

### ISSUE THREAD

**Issue by UTMark** [microsoft/chat-copilot#842]
**Describe the bug**
After uploading a file, citation will always be referenced with a relevance score despite not actually referencing it.

**To Reproduce**
1. Upload file
2. Talk to Copilot
3. See citation

**Expected behavior**
Should only show citation card when referencing source file. 

**Screenshots**
![image](https://github.com/microsoft/chat-copilot/assets/153741058/8d2090c0-3575-4ae9-8101-1a62642eac74)
_*screenshot from 0.9.0 but same thing happens with 0.9.1_

**Platform**
 - OS: Windows, Azure
 - IDE: VS Code

**Additional context**
Happens for versions 0.9.0, 0.9.1 (tags)
On rare occasion I have had it not reference any document as expected.
Document uploaded was in docx format and also used pdf in other threads, same behavior

**Azure configuration**
Kernel Memory Storage Type: Azure Blobs
Orchestration Type: Simple Queues
Vector DB Ingestion Type: In Process
Vector DB Retrieval Type: Azure Cognitive Search

**Comment by Riya-Chaudhary-dev** [comment]
I have had the same issue. When I tried to redeploy the services, the ai search no longer works(no index is being created)

**Comment by glahaye** [comment]
Could this be related to #877?

**Comment by YJasonY** [comment]
We've encountered this issue as well - it's not related to the setup, but with the workflow itself.

The citations shown on the UI are generated by (in this case) Azure AI Search, whereas the actual bot response comes from AI. This leads to the scenarios as described, where the bot generates a response without needing to use the content in the citation(s).

We got around this by adding another process to have the AI 'validate' whether or not their response actually used any of the citation content before deciding whether to display the citations on the client-side. 

Basically:
- don't display citations while the AI is generating an answer
- wait until the AI finishes generating the response (the 'response')
- extract the citation content into a string (the 'citations')
- create a new prompt template with a system prompt like "does the following 'response' use any content from the 'citations'? respond with 'yes' or 'no' only."
- parse the response and show/hide citations accordingly

Works reasonably well for us, but curious if anyone has another solution.
